### PR TITLE
Improve tests to explicitly distinguish old and new robonode behaviour at `robonode-client`

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

File: crates/robonode-client/src/enroll.rs [hunk 1]
```diff
@@ -279,8 +292,14 @@ mod tests {
...
-            let response =
-                ResponseTemplate::new(case.0).set_body_json(mkerr_containing_blob(case.1, "blob"));
+            let response = match case.3 {
+                ResponseIncludesBlob::Yes => {
+                    ResponseTemplate::new(case.0).set_body_json(mkerr(case.1, "scan result blob"))
+                }
+                ResponseIncludesBlob::No => {
+                    ResponseTemplate::new(case.0).set_body_json(mkerr_before_2023_05(case.1))
+                }
+            };
```
Comment (MOZGIII) [inline crates/robonode-client/src/enroll.rs]:
```suggestion
            let response_body = match case.3 {
                ResponseIncludesBlob::Yes => mkerr(case.1, "scan result blob"),
                ResponseIncludesBlob::No => mkerr_before_2023_05(case.1),
            };
            let response = ResponseTemplate::new(case.0).set_body_json(response_body);
```

Comment (dmitrylavrenov) [reply crates/robonode-client/src/enroll.rs]: https://github.com/humanode-network/humanode/pull/1271/commits/bcc2e966c77f09ab977245e0e4a6383402b9dcda

File: crates/robonode-client/src/authenticate.rs [hunk 2]
```diff
@@ -272,55 +272,65 @@ mod tests {
...
             };
 
-            let response =
-                ResponseTemplate::new(case.0).set_body_json(mkerr_containing_blob(case.1, "blob"));
+            let response_body = match response_includes_blob {
+                ResponseIncludesBlob::Yes => mkerr(body, Some("scan result blob")),
+                ResponseIncludesBlob::No => mkerr(body, None),
+            };
+
+            let response = ResponseTemplate::new(status_code).set_body_json(response_body);
```
Comment (MOZGIII) [inline crates/robonode-client/src/authenticate.rs:333]:
```suggestion
            let response_body = mkerr(body, match response_includes_blob {
                ResponseIncludesBlob::Yes => Some("scan result blob"),
                ResponseIncludesBlob::No => None,
            });

            let response = ResponseTemplate::new(status_code).set_body_json(response_body);
```

File: crates/robonode-client/src/authenticate.rs [hunk 3]
```diff
@@ -272,55 +272,65 @@ mod tests {
...
             (
                 StatusCode::BAD_REQUEST,
                 "MY_ERR_CODE",
+                ResponseIncludesBlob::No,
                 AuthenticateError::UnknownCode("MY_ERR_CODE".to_owned()),
             ),
         ];
 
-        for case in cases {
+        for (status_code, body, response_includes_blob, expected_error) in cases {
```
Comment (MOZGIII) [inline crates/robonode-client/src/authenticate.rs:320]:
```suggestion
        for (http_code, error_code, response_includes_blob, expected_error) in cases {
```

Comment (dmitrylavrenov) [reply crates/robonode-client/src/authenticate.rs]: https://github.com/humanode-network/humanode/pull/1271/commits/88d0bc2810c5a2d10dfac7c80568574e0e69fa85

File: crates/robonode-client/src/authenticate.rs (hunk 2 above)
Comment (dmitrylavrenov) [reply crates/robonode-client/src/authenticate.rs]: https://github.com/humanode-network/humanode/pull/1271/commits/5a2b11335d2a13ab14075fa48dcdd43f254f417b

**Review by MOZGIII — APPROVED** [review]

**Pull request merged**